src/core/src/node.cpp

```diff
@@ -472,8 +467,8 @@ ov::descriptor::Tensor& ov::Node::get_output_tensor(size_t i) const {
 
 ov::descriptor::Tensor& ov::Node::get_input_tensor(size_t i) const {
     OPENVINO_ASSERT(i < m_inputs.size(), idx_txt, i, out_of_range_txt);
-    descriptor::Input input = m_inputs[i];
-    return input.get_tensor();
+    auto& input = m_inputs[i];
+    return input.get_output().get_tensor();
```

Comment:
Why this had to be changed? `input.get_tensor()` wouldn't work?

Reply:
The first line it was just to remove unnecessary copy which was used to make a little hack (similar to `const_cast`) to remove consents from `input` as this function is `const`.
Without copy the connected node output tensor has to be returned which is not constant and it give better description what is returned . 

This is maybe separate topic but maybe `get_input_tensor` tensor should return const descriptor to avoid modification of descriptors which not belongs to this Node. The Node's inputs are not tensors owners.